src/options.cc

```diff
@@ -2,22 +2,27 @@
 
 #include "options.h"
 
-Options::Options(const char* file_name) {
+void Options::loadOptionsFromFile(const char* filename) {
 	NginxConfigParser parser;
-	parser.Parse(file_name, &config);
-}
+	NginxConfig config;
+	parser.Parse(filename, &config);
 
-unsigned short Options::getPort() {
 	for (unsigned int i =0; i < config.statements_.size();i++) {
+
+		// Sets the port
 		if (config.statements_[i]->tokens_[0] == "server") {
 			std::shared_ptr<NginxConfigStatement> temp_config = config.statements_[i];
 			for (unsigned int j = 0; j < temp_config->child_block_->statements_.size(); j++) {
 				if(temp_config->child_block_->statements_[j]->tokens_[0] == "listen") {
 					std::string port = temp_config->child_block_->statements_[j]->tokens_[1];
-					return (unsigned short) std::stoi(port);
+					this->port = (unsigned short) std::stoi(port);
 				}
 			}
 		}
 	}
-    return -1;
 }
+
+unsigned short Options::getPort() {
+	//TODO check port here
+	return this->port;
+};
```

Comment:
Since we're using a struct, we should be able to get rid of `getPort` entirely, and have the caller (in main) just do something like: 
```c++
Options opt;
bool parseSuccess = opt.loadOptionsFromFile(argv[1]);
if ( ! parseSuccess) {
   /* put some error & exit here */
}

// if we've gotten to here, we know opt.port was loaded correctly and is a valid port
unsigned short port_num = opt.port;
```
(basically just like you're already doing)